src/main/java/com/ghostipedia/cosmiccore/client/renderer/machine/HellFireFoundryPartRender.java

```diff
@@ -0,0 +1,104 @@
+                                       implements IControllerModelRenderer {
+
+    // spotless:off
+    public static final Codec<HellFireFoundryPartRender> CODEC = RecordCodecBuilder.create(instance -> instance.group(
+            BlockState.CODEC.fieldOf("casing_block").forGetter(HellFireFoundryPartRender::getCasing)
+    ).apply(instance, HellFireFoundryPartRender::new));
+    public static final DynamicRenderType<WorkableElectricMultiblockMachine, HellFireFoundryPartRender> TYPE = new DynamicRenderType<>(CODEC);
+    // spotless:on
+
+    private final BlockState iesniumBlock;
+    @Getter
+    private final BlockState casing;
+    private BakedModel iesniumBlockModel;
+    private BakedModel casingModel;
+
+    public HellFireFoundryPartRender(BlockState casing) {
+        this.iesniumBlock = OccultismBlocks.IESNIUM_BLOCK.get().defaultBlockState();
+        this.casing = casing;
+        ModelUtils.registerBakeEventListener(event -> {
+            this.iesniumBlockModel = event.getModels().get(BlockModelShaper.stateToModelLocation(this.iesniumBlock));
+            this.casingModel = event.getModels().get(BlockModelShaper.stateToModelLocation(this.casing));
+        });
+    }
+
+    @Override
+    public DynamicRenderType<WorkableElectricMultiblockMachine, HellFireFoundryPartRender> getType() {
+        return TYPE;
+    }
+
+    @Override
+    public void renderPartModel(List<BakedQuad> quads, IMultiController controller,
+                                IMultiPart part, Direction frontFacing, @Nullable Direction side,
+                                RandomSource rand, ModelData modelData, @Nullable RenderType renderType) {
+        BlockPos partPos = part.self().getPos();
+        MultiblockControllerMachine machine = controller.self();
+        BakedModel model = part instanceof SoulHatchPartMachine ? iesniumBlockModel : casingModel;
+        emitQuads(quads, model, machine.getLevel(), partPos, casing, side, rand, modelData, renderType);
+    }
+
+    private static void emitQuads(List<BakedQuad> quads, @Nullable BakedModel model,
+                                  BlockAndTintGetter level, BlockPos pos, BlockState state,
+                                  @Nullable Direction side, RandomSource rand,
+                                  ModelData modelData, @Nullable RenderType renderType) {
+        if (model == null) return;
+        modelData = model.getModelData(level, pos, state, modelData);
+        quads.addAll(model.getQuads(state, side, rand, modelData, renderType));
+    }
+
+    @Override
+    public void render(WorkableElectricMultiblockMachine machine, float partialTick, PoseStack poseStack,
+                       MultiBufferSource buffer, int packedLight, int packedOverlay) {}
+
+    @Override
+    public boolean shouldRender(WorkableElectricMultiblockMachine machine, Vec3 cameraPos) {
+        return false;
+    }
```

Comment:
What does a dynamic renderer with an overridden empty render function and shouldrender->false even do?